### DISCUSSION TITLE
Integrate `commitizen` to manage project versions/tags?

### DISCUSSION THREAD

**Discussion by baggiponte** [astral-sh/rye#388] [Ideas]
I have been using [commitizen](https://commitizen-tools.github.io/commitizen/) extensively to manage versions. I am fine with that setup (I use PDM) but I found that a lot of friends who use poetry use plugins like [`poetry-version-plugin`](https://github.com/tiangolo/poetry-version-plugin) by tiangolo to manage version bumps from commit history. There is a [similar manual plugin](https://github.com/carstencodes/pdm-bump) for PDM that IIUC does not manage tags automatically. Perhaps a "one true tool" should do this too?

**Comment by ischaojie** [comment]
In fact, you can use `rye version --bump <major/minor/patch>` to manage the version manually. (But there are still some minor limitations.
